File: Triatlon-Java/TriatlonService/src/main/java/service/TriatlonServiceInterface.java
package service;

import domain.*;

import java.util.List;
import java.util.Map;

public interface TriatlonServiceInterface {

/*
    Arbitru getAccount(String username, String parola, TriatlonObserverInterface client) throws Exception;
*/
    Arbitru login(ArbitruDTO arbitru, TriatlonObserverInterface client) throws Exception;

    void logout(Arbitru arbitru, TriatlonObserverInterface client) throws Exception;
    List<ParticipantDTO> getParticipantiCuPunctaj(TriatlonObserverInterface client) throws Exception;
    Iterable<Proba> findAllProbe() throws Exception;
    void addInscriere(Inscriere inscriere) throws Exception;
    List<ParticipantDTO> getParticipantiDeLaProba(Proba proba, TriatlonObserverInterface client ) throws Exception;

}

File: Triatlon-C#/TriatlonService/TriatlonServiceInterface.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TriatlonModel;

namespace TriatlonServices
{
    public interface TriatlonServiceInterface
    {
        //Arbitru GetAccount(string username, string password);
        Arbitru GetAccount(String username, String parola);
        Arbitru login(ArbitruDTO arbitru, TriatlonObserverInterface client);

        void logout(Arbitru arbitru, TriatlonObserverInterface client);
        List<ParticipantDTO> GetParticipantiCuPunctaj(TriatlonObserverInterface client);


        IEnumerable<Proba> FindAllProbe(TriatlonObserverInterface client);

        void AddInscriere(Inscriere inscriere, TriatlonObserverInterface client);

        List<ParticipantDTO> GetParticipantiDeLaProba(Proba proba, TriatlonObserverInterface client);

        IEnumerable<ParticipantDTO> GetParticipantiDTO(TriatlonObserverInterface client);

        IEnumerable<ParticipantDTO> GetParticipantiDTOprobaActuala(Proba proba, TriatlonObserverInterface client);

    }
}
